src/Algebra/Ordered/Definitions.agda

```diff
@@ -0,0 +1,41 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of functions on ordered structures
+------------------------------------------------------------------------
+
+-- The contents of this module should be accessed via `Algebra.Ordered`.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Relation.Binary.Core
+
+module Algebra.Ordered.Definitions
+  {a ℓ} {A : Set a}   -- The underlying set
+  (_≤_ : Rel A ℓ)     -- The underlying order
+  where
+
+open import Algebra.Core
+open import Function.Base using (flip)
+open import Data.Product using (_×_)
+
+Monotonic₁ : Op₁ A → Set _
+Monotonic₁ f = f Preserves _≤_ ⟶ _≤_
```

Comment:
Hmm I'm now wondering if these should go in `Relation.Binary.Definitions` instead of here?

Reply:
The reason I put them here is that they involve binary/unary operations from `Algebra`. There are analogous `Congruence` definitions in `Algebra.Definitions`:
https://github.com/agda/agda-stdlib/blob/e0e5db48582da275ce5f1fa747505795ee0b714c/src/Algebra/Definitions.agda#L27-L31

I figured that, if these live in the `Algebra` hierarchy (rather than `Relation.Binary`), then so should `Monotonic` and `Antitonic`.

Reply:
So actually `_Preserves _⟶ _` and friends are from the `Relation.Binary` hierarchy. I think the notion of monotonic and antitonic functions make sense for _any_ functions, rather than just those that are used in an algebraic setting right? For example `f x = x ::_` is monotonic over the lexicographic ordering over lists.

Reply:
That's true, but it's equally true for `Congruent`. I mean the only difference between `Preserves`, `Monotone` and `Congruent` is how they are intended to be used. Maybe adding all these anti/mono/whathever definitions is overkill and we should just use `Preserves` directly? (That's what I did originally, but then I figured I should probably mirror the definition of `Congruent`.)

Reply:
That's true about `Congruent`. Perhaps having it in `Algebra.Definitions` instead of `Relation.Binary.Definitions` is a mistake. Nonetheless, I would really like these new monotonic definitions to live in the latter. 

Another argument is that if we ever want to add a new ordered definition that used both the order and the equality relation, then this module's current module-level arguments wouldn't allow that.

Reply:
OK, so just to double check before making the changes:

 1. I'll move **all** the definitions from `Algebra.Definitions.Ordered` to `Relation.Binary.Definitions` then. 
 2. I'll drop the use of `Op` from `Algebra.Definitions` (to make the definition generic, and avoid cycles in the module imports).
 3. I'll keep `{Mono,Andt}tonic` even though they will be essentially aliases of `Preserves`.

Agreed?